File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFilterState.java
package fr.free.nrw.commons.nearby;

public class NearbyFilterState {
    private boolean existsSelected;
    private boolean needPhotoSelected;

    private static NearbyFilterState nearbyFılterStateInstance;

    /**
     * Define initial filter values here
     */
    private NearbyFilterState() {
        existsSelected = false;
        needPhotoSelected = true;
    }

    public static NearbyFilterState getInstance() {
        if (nearbyFılterStateInstance == null) {
            nearbyFılterStateInstance = new NearbyFilterState();
        }
        return nearbyFılterStateInstance;
    }

    public static void setExistsSelected(boolean existsSelected) {
        getInstance().existsSelected = existsSelected;
    }

    public static void setNeedPhotoSelected(boolean needPhotoSelected) {
        getInstance().needPhotoSelected = needPhotoSelected;
    }

    public boolean isExistsSelected() {
        return existsSelected;
    }

    public boolean isNeedPhotoSelected() {
        return needPhotoSelected;
    }

}

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFilterState.kt
package fr.free.nrw.commons.nearby

object NearbyFilterState {
    var isExistsSelected: Boolean = true
        private set
    var isNeedPhotoSelected: Boolean = true
        private set
    var isWlmSelected: Boolean = true
        private set
    val checkBoxTriState: Int = -1
    var selectedLabels: List<Label> = emptyList()
        private set

    fun setSelectedLabels(labels: List<Label>) {
        selectedLabels = labels
    }

    fun setExistsSelected(existsSelected: Boolean) {
        isExistsSelected = existsSelected
    }

    fun setNeedPhotoSelected(needPhotoSelected: Boolean) {
        isNeedPhotoSelected = needPhotoSelected
    }

    fun setWlmSelected(wlmSelected: Boolean) {
        isWlmSelected = wlmSelected
    }
}
